src/daft-dsl/src/functions/list/sum.rs

```diff
@@ -0,0 +1,34 @@
+use crate::Expr;
+use daft_core::{datatypes::Field, schema::Schema, series::Series};
+
+use common_error::{DaftError, DaftResult};
+
+use super::super::FunctionEvaluator;
+
+pub(super) struct SumEvaluator {}
+
+impl FunctionEvaluator for SumEvaluator {
```

Comment:
We can probably leverage a macro for all the List FunctionEvaluators

Reply:
The `to_field` logics are different except for `max` and `min`, would it make sense to have a macro?